src/translate.py

```diff
@@ -3726,28 +3762,29 @@ def make_arg(offset: int, type: Type) -> PassedInArg:
         assert offset % 4 == 0
         return PassedInArg(offset, copied=False, stack_info=stack_info, type=type)
 
-    c_fn: Optional[CFunction] = None
-    known_params = False
-    variadic = False
-    return_type = Type.any_reg()
     if typemap and function.name in typemap.functions:
-        c_fn = typemap.functions[function.name]
-        if c_fn.ret_type is not None:
-            return_type = type_from_ctype(c_fn.ret_type, typemap)
-        if c_fn.is_variadic:
-            stack_info.is_variadic = True
-        if c_fn.params is not None:
-            abi_slots, possible_regs = function_abi(c_fn, typemap, for_call=False)
-            for slot in abi_slots:
-                stack_info.add_known_param(slot.offset, slot.name, slot.type)
-                if slot.reg is not None:
-                    initial_regs[slot.reg] = make_arg(slot.offset, slot.type)
-            for reg in possible_regs:
-                offset = 4 * int(reg.register_name[1])
-                initial_regs[reg] = make_arg(offset, Type.any_reg())
-            known_params = True
-
-    if not known_params:
+        fn_type = type_from_ctype(typemap.functions[function.name].type, typemap)
+        fn_decl_provided = True
+    else:
+        fn_type = Type.function()
+        fn_decl_provided = False
+    fn_type.unify(function_sym.type)
```

Comment:
nit: I'd inline the definition of function_sym here to reduce the number of live variables, since it's only used once